api/src/services/authentication.ts

```diff
@@ -58,31 +57,33 @@ export class AuthenticationService {
 
 		const { email, password, ip, userAgent, otp } = options;
 
-		const hookPayload = omit(options, 'password', 'otp');
-
-		const user = await database
+		let user = await database
 			.select('id', 'password', 'role', 'tfa_secret', 'status')
 			.from('directus_users')
 			.whereRaw('LOWER(??) = ?', ['email', email.toLowerCase()])
 			.first();
 
-		await emitter.emitAsync('auth.login.before', hookPayload, {
+		const updatedUser = await emitter.emitAsync('auth.login.before', options, {
```

Comment:
Last thing: We'll have to make sure this returns the correct thing. IIRC the emitter returns an array of payloads, as multiple hooks can modify the same payload in order of registration. We can use the following to merge those back into the final augmented object:

```js
const newPayload = hooksResult.reverse().reduce((val, acc) => merge(acc, val))
```

Reply:
I am making some tests locally and I think we should not use the `reverse` because I think the last registered hook should have priority on changes. This means on merge, last hook should have priority over others. Don't you agree?

Reply:
I ended up using only the `reduce` since I think it makes more sense. But we can replace with `reduceRight` if you think it's better.

Reply:
I'll have to double check why it is like that in the first place. Regardless, I wanna keep it the same in both places, so if it makes most sense to drop `reverse`, we should do it everywhere 👍🏻

Reply:
So, should I put `reverse` in this PR and open another which removes `reverse` from hook handling?
I am only seeing the other cases here:
https://github.com/directus/directus/blob/31113a3087973237f52883199afeaed4540713c7/api/src/services/items.ts#L96
https://github.com/directus/directus/blob/31113a3087973237f52883199afeaed4540713c7/api/src/services/items.ts#L375
https://github.com/directus/directus/blob/31113a3087973237f52883199afeaed4540713c7/api/src/controllers/not-found.ts#L19

Reply:
We can do it all in this PR, that's fine. I just want to make sure it's consistent everywhere, and the expected behavior everywhere 👍🏻

Reply:
No problem. Here it is 5885b69ecd29ca2a4e94e00a7a4a1f912c6e76a2 ;)